src/portgraph.rs

```diff
@@ -1391,9 +1417,14 @@ pub enum LinkError {
     /// The port offset is invalid.
     #[error("unknown port offset {} in node {node:?} in direction {:?}", offset.index(), offset.direction())]
     UnknownOffset { node: NodeIndex, offset: PortOffset },
-    /// The port cannot be linked in this direction.
-    #[error("port {port:?} had an unexpected direction {dir:?} during a link operation")]
-    UnexpectedDirection { port: PortIndex, dir: Direction },
+    /// The ports have the same direction so they cannot be linked.
+    #[error("Cannot link ports with direction {dir_from:?} and {dir_to:?}. In ports {port_from:?} and {port_to:?}")]
```

Comment:
Oh, and - you only need to give the direction once - the two directions you print out here will always be the same!

Reply:
And drop `from`/`to` naming as well I guess.